schemaregistry/schemaregistry.yaml

```diff
@@ -0,0 +1,313 @@
+openapi: 3.0.0
+info:
+  title: Cloud Native Data Schema Registry
+  version: 0.1
+paths:
+  /schemagroups:
+    get:
+      summary: 'Get list of schema groups'
+      description: 'Get all schema groups in namespace.'
+      operationId: getGroups
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items: 
+                  type: string
+      tags:
+      -  'groups'
+  /schemagroups/{group-name}:
+    parameters:
+      - name: group-name
+        in: path
+        description: schema group
+        required: true
+        schema:
+          type: string
+    get:
+      summary: Get schema group
+      description: Get schema group description in registry namespace.
+      operationId: getGroup
+      responses:
+        '200': 
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SchemaGroup'
+        '404':
+          description: Specified group not found
+      tags:
+      - 'groups'
+    put:
+      summary: Create schema group
+      description: Create schema group with specified format format in registry namespace.
+      operationId: createGroup 
+      requestBody:
+        description: schema group description
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SchemaGroup'
+      responses:
+        '201':
+          description: Created
+          headers:
+            Location:
+              schema:
+                type: string
+        '409':
+          description: Schema group already exists
+      tags:
+      -  'groups'
+    delete:
```

Comment:
Should DELETE actually be allowed? Did you consider a "decomission" option or "soft delete" alternatively? That'd e.g. prohibit to produce new events referencing such schema taken out of business, but existing events could still be decoded.

Reply:
Yes, in general deletes are problematic and should IMO be either discouraged or prevented.  Immutability has been a useful property in our registry for schema versions - we can deprecate and disable versions.  And I wish we had done the same for the entire schema (deprecate/disable) rather than allow deletes.